andela/ah-scorpion
# #159053995 Add endpoints for search with custom filters

**What does this PR do?**
- Allow users to make custom searches for articles

**Description of tasks completed**
- Create endpoints for different search filters such as tags, authors and content
- Write unit tests for the above functionality


**How can this PR be manually tested**
- Git pull the source code from ft-159053995-Search-Functionality-with-custom-filters
- Copy the variables from `sample.env` and put the correct parameters in `.env` file
- Run `source .env` on your terminal to export environmental variables and activate virtual env
- Run `python manage.py migrate` to apply recent migrations
- Run `pip install -r requirements` to install latest dependencies
- Run `python manage.py runserver` to start the project
- If not already done, create a new user account and login
- Create an article as shown in the previous instructions
- Make a GET request to the following endpoint `api/v1/articles/?author__username=<username>` replacing the `<username>` with the username. The data should be articles created by that username
- Make a GET request to the following endpoint `api/v1/articles/?tagList=<anytag>` replacing the `<tag>` with the tag you had used. The data should be articles created with that tag
- Make a GET request to the following endpoint `api/v1/articles/?body=<any text in the body>`  The data should be articles with the text content searched
- Make a GET request to the following endpoint `api/v1/articles/?title=<title>` replacing the `<title>` with the title you had used. The data should be articles with the same title



**Relevant Stories**
[159053995](https://www.pivotaltracker.com/n/projects/2184741/stories/159053995)
